src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -606,6 +613,11 @@ public MetadataEnumeration<SnapshotSummary> retrieveSnapshots(
       throw new IllegalArgumentException(
           "Failed to convert snapshot request tags list to SQL array", e);
     }
+    if (!ListUtils.emptyIfNull(duosIds).isEmpty()) {
```

Comment:
You can use `Objects` to do this. In general I think we should avoid third party libraries if the JDK itself has an equivalent API.
```suggestion
    if (!Objects.requireNonNullElse(duosIds, List.of()).isEmpty()) {
```

Although personally I think this kind of form is less readable than comparing against `null` directly
```suggestion
    if (duosIds != null && !duosIds.isEmpty()) {
```

Reply:
Also how can a `null` value for `duosIds` arrive here? Better to validate the data at the source rather than do it here IMO.